sipa/blueprints/register.py

```diff
@@ -0,0 +1,283 @@
+          'error')
+    logger.critical(
+        'Backend error: %s', ex.backend_name,
+        extra={'data': {'exception_args': ex.args}},
+        exc_info=True,
+    )
+
+
+@bp_register.route("/identify", methods=['GET', 'POST'])
+@register_redirect
+def identify(reg_state: RegisterState):
+    form = RegisterIdentifyForm()
+
+    suggest_skip = False
+    if form.validate_on_submit():
+        reg_state.first_name = form.first_name.data
+        reg_state.last_name = form.last_name.data
+        reg_state.birthdate = form.birthdate.data
+        reg_state.no_swdd_tenant = form.no_swdd_tenant.data
+        previous_dorm = backends.get_dormitory(form.previous_dorm.data)
+        reg_state.previous_dorm = previous_dorm.display_name if previous_dorm else None
+
+        if form.no_swdd_tenant.data or 'skip_verification' in request.form:
+            reg_state.skipped_verification = True
+            return goto_step('data')
+
+        try:
+            match = api.match_person(form.first_name.data, form.last_name.data,
+                                     form.birthdate.data, form.tenant_number.data,
+                                     reg_state.previous_dorm)
+            reg_state.move_in_date = match.begin
+            reg_state.room_id = match.room_id
+            reg_state.building = match.building
+            reg_state.room = match.room
+            reg_state.tenant_number = form.tenant_number.data
+            return goto_step('room')
+        except PycroftApiError as e:
+            if e.code == 'user_exists':
+                flash(gettext(
+                    'Zu den von dir angegebenen Daten existiert bereits eine Mitgliedschaft. '
+                    'Bitte wähle aus, in welchem Wohnheim du vorher gewohnt hast.'), 'error')
+            elif e.code == 'similar_user_exists':
+                flash(gettext('Für den dir zugeordneten Raum gibt es bereits ein Konto. '
+                              'Falls du denkst, dass es sich dabei um einen Fehler handelt, '
+                              'kannst du den entsprechenden Button klicken. Die Verifizierung wird '
+                              'dann später manuell durchgeführt.'), 'error')
+                suggest_skip = True
+            elif e.code == 'no_room_for_tenancies':
+                reg_state.tenant_number = form.tenant_number.data
+                return goto_step('data')
+            else:
+                flash(gettext(
+                    'Die Verifizierung deiner Daten mit dem Studentenwerk Dresden ist fehlgeschlagen. '
+                    'Bitte überprüfe, dass du die exakt selben Daten, wie beim Studentenwerk Dresden '
+                    'bzw. wie auf deinem Mietvertrag, angegeben hast. '
+                    'Um die Verifizierung zu überspringen, kannst du den entsprechenden Button '
+                    'klicken. Die Verifizierung wird dann später manuell durchgeführt.'), 'error')
```

Comment:
Quite a wall of text, the process is complicated enough anyway.
English suggestion:

> We could not match your data to a contract. Please check that it precisely matches the spelling in your rental agreement. Alternatively, you can choose to skip the automatic verification for now.

This is shorter and communicates everything necessary.